src/tds.rs

```diff
@@ -78,8 +80,20 @@ impl Context {
         self.last_meta.as_ref().map(Arc::clone)
     }
 
-    pub fn packet_size(&self) -> usize {
-        self.packet_size.load(Ordering::SeqCst) as usize
+    pub fn packet_size(&self) -> u32 {
+        self.packet_size.load(Ordering::SeqCst)
+    }
+
+    pub fn set_packet_size(&self, new_size: u32) {
+        self.packet_size.store(new_size, Ordering::SeqCst);
+    }
+
+    pub fn transaction_id(&self) -> u64 {
+        self.transaction_id.load(Ordering::SeqCst)
+    }
+
+    pub fn set_transaction_id(&self, id: u64) {
+        self.transaction_id.store(id, Ordering::SeqCst);
```

Comment:
My only worry is that here multiple threads could be setting the transaction id on the client at the same time. But seeing token_env_change.rs it looks like this piece of state is server-driven and per-connection, right?

Reply:
Like you said in your previous comment.

Reply:
The atomics are here only for ergonomic reasons. A connection is always single-threaded in every SQL database, so there is no way of threads being able to mess with the ids.

As you see the `Client` takes `&mut self` always, which forces this to be single-threaded (or at least sync with a `Mutex`).